laws/src/main/scala/algebra/laws/LatticeLaws.scala

```diff
@@ -5,111 +5,15 @@ import algebra.lattice._
-    "absorption" -> forAll { (x: A, y: A) =>
-      (A.join(x, A.meet(x, y)) ?== x) && (A.meet(x, A.join(x, y)) ?== x)
-    }
-  )
-
-  def distributiveLattice(implicit A: DistributiveLattice[A]) = new LatticeProperties(
-    name = "distributiveLattice",
-    parents = Seq(lattice),
-    join = Some(semilattice(A.joinSemilattice)),
-    meet = Some(semilattice(A.meetSemilattice)),
-    "distributive" -> forAll { (x: A, y: A, z: A) =>
-      (A.join(x, A.meet(y, z)) ?== A.meet(A.join(x, y), A.join(x, z))) &&
-        (A.meet(x, A.join(y, z)) ?== A.join(A.meet(x, y), A.meet(x, z)))
-    }
-  )
-
-  def boundedJoinSemilattice(implicit A: BoundedJoinSemilattice[A]) = new LatticeProperties(
-    name = "boundedJoinSemilattice",
-    parents = Seq(joinSemilattice),
-    join = Some(boundedSemilattice(A.joinSemilattice)),
-    meet = None
-  )
+trait LatticeLaws[A] extends JoinSemilatticeLaws[A] with MeetSemilatticeLaws[A] {
+  override implicit def S: Lattice[A]
 
-  def boundedMeetSemilattice(implicit A: BoundedMeetSemilattice[A]) = new LatticeProperties(
-    name = "boundedMeetSemilattice",
-    parents = Seq(meetSemilattice),
-    join = None,
-    meet = Some(boundedSemilattice(A.meetSemilattice))
-  )
-
-  def boundedJoinLattice(implicit A: Lattice[A] with BoundedJoinSemilattice[A]) = new LatticeProperties(
-    name = "boundedJoinLattice",
-    parents = Seq(boundedJoinSemilattice, lattice),
-    join = Some(boundedSemilattice(A.joinSemilattice)),
-    meet = Some(semilattice(A.meetSemilattice))
-  )
-
-  def boundedMeetLattice(implicit A: Lattice[A] with BoundedMeetSemilattice[A]) = new LatticeProperties(
-    name = "boundedMeetLattice",
-    parents = Seq(boundedMeetSemilattice, lattice),
-    join = Some(semilattice(A.joinSemilattice)),
-    meet = Some(boundedSemilattice(A.meetSemilattice))
-  )
-
-  def boundedLattice(implicit A: BoundedLattice[A]) = new LatticeProperties(
-    name = "boundedLattice",
-    parents = Seq(boundedJoinSemilattice, boundedMeetSemilattice, lattice),
-    join = Some(boundedSemilattice(A.joinSemilattice)),
-    meet = Some(boundedSemilattice(A.meetSemilattice))
-  )
-
-  def boundedDistributiveLattice(implicit A: BoundedDistributiveLattice[A]) = new LatticeProperties(
-    name = "boundedLattice",
-    parents = Seq(boundedLattice, distributiveLattice),
-    join = Some(boundedSemilattice(A.joinSemilattice)),
-    meet = Some(boundedSemilattice(A.meetSemilattice))
-  )
-
-  class LatticeProperties(
-    val name: String,
-    val parents: Seq[LatticeProperties],
-    val join: Option[GroupProperties],
-    val meet: Option[GroupProperties],
-    val props: (String, Prop)*
-  ) extends RuleSet {
-    private val _m = meet map { "meet" -> _ }
-    private val _j = join map { "join" -> _ }
-
-    val bases = _m.toList ::: _j.toList
-  }
+  def absorption(x: A, y: A)(implicit E: Eq[A]): IsEq[Boolean] =
+    (E.eqv(S.join(x, S.meet(x, y)), x) && E.eqv(S.meet(x, S.join(x, y)), x)) <-> true
+}
```

Comment:
I would split the absorption law in two pieces, so that the return type is `IsEq[A]`